vision/google/cloud/vision/annotations.py

```diff
@@ -0,0 +1,130 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Annotations management for Vision API responses."""
+
+
+from google.cloud.vision.entity import EntityAnnotation
+from google.cloud.vision.face import Face
+from google.cloud.vision.feature import FeatureTypes
+from google.cloud.vision.color import ImagePropertiesAnnotation
+from google.cloud.vision.safe import SafeSearchAnnotation
+
+
+_REVERSE_TYPES = {
+    FeatureTypes.FACE_DETECTION: 'faceAnnotations',
+    FeatureTypes.IMAGE_PROPERTIES: 'imagePropertiesAnnotation',
+    FeatureTypes.LABEL_DETECTION: 'labelAnnotations',
+    FeatureTypes.LANDMARK_DETECTION: 'landmarkAnnotations',
+    FeatureTypes.LOGO_DETECTION: 'logoAnnotations',
+    FeatureTypes.SAFE_SEARCH_DETECTION: 'safeSearchAnnotation',
+    FeatureTypes.TEXT_DETECTION: 'textAnnotations',
+}
+
+
+class Annotations(object):
+    """Annotation class for managing responses.
+
+    :type faces: list
+    :param faces: List of :class:`~google.cloud.vision.face.Face`.
+
+    :type properties: list
+    :param properties:
+        List of :class:`~google.cloud.vision.color.ImagePropertiesAnnotation`.
+
+    :type labels: list
+    :param labels: List of
+                   :class:`~google.cloud.vision.entity.EntityAnnotation`.
+
+    :type landmarks: list
+    :param landmarks: List of
+                      :class:`~google.cloud.vision.entity.EntityAnnotation.`
+
+    :type logos: list
+    :param logos: List of
+                  :class:`~google.cloud.vision.entity.EntityAnnotation`.
+
+    :type safe_searches: list
+    :param safe_searches:
+        List of :class:`~google.cloud.vision.safe.SafeSearchAnnotation`
+
+    :type texts: list
+    :param texts: List of
+                  :class:`~google.cloud.vision.entity.EntityAnnotation`.
+    """
+    def __init__(self, faces=None, properties=None, labels=None,
+                 landmarks=None, logos=None, safe_searches=None, texts=None):
+        self.faces = faces or []
+        self.properties = properties or []
+        self.labels = labels or []
+        self.landmarks = landmarks or []
+        self.logos = logos or []
+        self.safe_searches = safe_searches or []
+        self.texts = texts or []
+
+    @classmethod
+    def from_api_repr(cls, response):
+        """Factory: construct an instance of ``Annotations`` from a response.
+
+        :type response: dict
+        :param response: Vision API response object.
+
+        :rtype: :class:`~google.cloud.vision.annotations.Annotations`
+        :returns: An instance of ``Annotations`` with detection types loaded.
+        """
+        annotations = {}
+
+        for feature_type in response.keys():
+            annotations[feature_type] = []
```

Comment:
Instead of using this, use `setdefault` in your next loop (i.e. just delete this loop).